data_steward/cdr_cleaner/cleaning_rules/generate_derived_tables.py

```diff
@@ -0,0 +1,397 @@
+"""
+Create Derived Tables
+
+Until now, 'observation_period', 'drug_era, 'condition_era' were populated Pre-deid stages.
+But re-creating these tables post-deid will allow OHDSI tools to run on workbench more accurately.
+"""
+# Python imports
+import logging
+from datetime import datetime
+import os
+import json
+
+# Third party imports
+
+# Project imports
+import constants.cdr_cleaner.clean_cdr as cdr_consts
+from resources import validate_date_string
+from common import CONDITION_ERA, DRUG_ERA, OBSERVATION_PERIOD, JINJA_ENV
+from cdr_cleaner.cleaning_rules.base_cleaning_rule import BaseCleaningRule
+from resources import cdm_fields_path
+
+LOGGER = logging.getLogger(__name__)
+
+DELETION_QUERY = JINJA_ENV.from_string("""
+DELETE FROM `{{project_id}}.{{dataset_id}}.{{storage_table_name}}`
+WHERE TRUE
+""")
+
+POPULATE_OBS_PRD = JINJA_ENV.from_string("""
+INSERT INTO `{{project_id}}.{{dataset_id}}.{{storage_table_name}}` {{cols}}
+SELECT ROW_NUMBER() OVER (ORDER BY person_id) AS observation_period_id,
+person_id,
+observation_period_start_date,
+    CASE WHEN observation_period_end_date > DATE('{{ehr_cutoff_date}}') THEN '{{ehr_cutoff_date}}'
+    ELSE observation_period_end_date
+END AS observation_period_end_date,
+period_type_concept_id
+FROM (SELECT person_id,
+    MIN(observation_period_start_date) AS observation_period_start_date,
+    MAX(observation_period_end_date) AS observation_period_end_date,
+    44814725 AS period_type_concept_id
+    FROM (SELECT pt.person_id AS person_id,
+            MIN(vt.visit_start_date) AS observation_period_start_date,
+            MAX(vt.visit_end_date) AS observation_period_end_date
+        FROM `{{project_id}}.{{dataset_id}}.person` AS pt
+        JOIN `{{project_id}}.{{dataset_id}}.visit_occurrence` AS vt ON pt.person_id = vt.person_id
+        JOIN `{{project_id}}.{{dataset_id}}.visit_occurrence_ext` AS e ON vt.visit_occurrence_id = e.visit_occurrence_id
+            AND EXTRACT(YEAR FROM vt.visit_start_date) >= 1985
```

Comment:
These can be updated to "WHERE"
```suggestion
            WHERE EXTRACT(YEAR FROM vt.visit_start_date) >= 1985
```